clang/docs/ReleaseNotes.rst

```diff
@@ -146,6 +146,13 @@ Non-comprehensive list of changes in this release
 
 New Compiler Flags
 ------------------
+* ``-fverify-intermediate-code`` and it's complement ``-fno-verify-intermediate-code``.
+  Enables or disables verification of the generated LLVM IR.
+  It's strongly encouraged to enable this verification, as it can catch hard to
+  find code generation bugs.
```

Comment:
This sentence makes no sense to me. End users certainly shouldn't run with enabled IR verification. Developers already use IR verification by dint of using an assertion-enabled build.

Reply:
Does it really not make sense, or do you simply not agree we should recommend it?

I understand LLVM developers generally test the toolchain with runtime checks enabled.

But the sticking point for me is that I find it implausible that this is a satisfactory level of coverage, specially when you consider that there are some platforms that have significant quirks, but that are not well represented within the developers subset.

For an example, see https://github.com/llvm/llvm-project/issues/67937
We have been generating ill-formed IR for many years on the MSVC platform, for simple basic scenario that should be covered by any real world use case.

Reply:
> Does it really not make sense, or do you simply not agree we should recommend it?

Both? It doesn't make sense to "strongly recommend" that end users manually enable verification by passing `-fverify-intermediate-code`. If we think it's important, then we need to enable the option by default, not ask users to do so. (Of course, I am strongly opposed to actually enabling it by default.)

Reply:
Strongly recommending doing something that is opt-in, versus making that something opt-out, versus forcing it, all have different trade-offs, and I am not trying to make any blank statements here that any of them don't make sense.

I think keeping it opt-in, but recommending people do so, is sensible progressive political approach, especially when this choice can have financial costs on them :)

Reply:
FWIW, I am proposing this recommendation based on these numbers: https://llvm-compile-time-tracker.com/compare.php?from=634c0f8ea4acd0694d89c3e2d278e6bcb78b923f&to=49da7e1de6348f1c67a952e9cfea93abe920c4d5&stat=instructions:u

It 'feels' to me like to me that cost is high enough that we shouldn't force enable it for simplicity, not even make it opt-out, because will take folks by surprise, but it's not really high enough that I wouldn't feel comfortable strongly recommending they try it.

Reply:
I'd word it like "users can pass this to turn on extra verification to catch compiler bugs at the cost of extra compile time"; I agree that we shouldn't be recommending it since the cost/benefit doesn't seem worth it for production toolchains. It's useful for validating a toolchain when you update the toolchain, but probably not super impactful after that.

Reply:
@aeubanks I like that wording.

Reply:
Okay, sounds good, I just implemented these changes to the document :)